internal/configs/module_merge.go

```diff
@@ -265,6 +265,9 @@ func (r *Resource) merge(or *Resource, rps map[string]*RequiredProvider) hcl.Dia
 		if or.Managed.PreventDestroy != nil {
 			r.Managed.PreventDestroy = or.Managed.PreventDestroy
 		}
+
+		r.Managed.SkipDestroy = or.Managed.SkipDestroy
```

Comment:
Our usual rules for override files (which is what this function is handling) is that we only overwrite the based configuration if the argument is explicitly defined in the override file, and not if it's just omitted and left as the default. In other words, the default for an argument in an override file is "whatever the base file declared".

That is why, for example, `CreateBeforeDestroy` also has `CreateBeforeDestroySet` above. `PreventDestroy` also had `PreventDestroySet` until the recent change to make `PreventDestroy` a `hcl.Expression` field instead of a `bool` field; if you agree with my previous comment about making `SkipDestroy` also be `hcl.Expression` here then we can use nil to represent "unset" and so avoid the extra field for whether it's set or not.